src/auth/providers.js

```diff
@@ -217,15 +233,19 @@ function apiKeyProvider(auth) {
      * @param {String} key the key for authentication
      * @returns {Promise} a promise that resolves when authentication succeeds.
      */
-    authenticate: key => {
+    authenticate: (key, link) => {
       const deviceId = auth.getDeviceId();
       const device = auth.getDeviceInfo(deviceId, !!auth.client && auth.client.clientAppID);
       const fetchArgs = common.makeFetchArgs(
         'POST',
         JSON.stringify({ 'key': key, 'options': { device } })
       );
       fetchArgs.cors = true;
-      return fetch(`${auth.rootUrl}/${loginRoute}`, fetchArgs)
+      if (link) {
```

Comment:
[nit] Can we refactor this into another function like `fetchArgsWithLink`? Probably needs to be a method on auth though.